core/src/test/java/org/apache/carbondata/core/localdictionary/TestLocalDictionaryGenerator.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.carbondata.core.localdictionary;
+
+import java.util.Arrays;
+
+import org.apache.carbondata.core.constants.CarbonCommonConstants;
+import org.apache.carbondata.core.localdictionary.exception.DictionaryThresholdReachedException;
+import org.apache.carbondata.core.localdictionary.generator.ColumnLocalDictionaryGenerator;
+import org.apache.carbondata.core.localdictionary.generator.LocalDictionaryGenerator;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestLocalDictionaryGenerator {
+
+  @Test
+  public void testColumnLocalDictionaryGeneratorWithValidDataWithinThreshold() {
+    LocalDictionaryGenerator generator = new ColumnLocalDictionaryGenerator(1000);
```

Comment:
@ravipesala @akashrn5 The constructor ColumnLocalDictionaryGenerator(int) has been modified to constructor ColumnLocalDictionaryGenerator(int, int), please update these issues.

Reply:
@zzcclp pr is raised #2428